solidity/contracts/liquidity-protection/LiquidityProtectionSettings.sol

```diff
@@ -237,7 +210,21 @@ contract LiquidityProtectionSettings is ILiquidityProtectionSettings, AccessCont
      * @return true if the given pool is whitelisted, false otherwise
      */
     function isPoolWhitelisted(IConverterAnchor _poolAnchor) external view override returns (bool) {
-        return poolWhitelistIndices[_poolAnchor].isValid;
+        return _poolWhitelist.contains(address(_poolAnchor));
+    }
+
+    /**
+     * @dev returns pools whitelist
+     *
+     * @return pools whitelist
+     */
+    function poolWhitelist() external view returns (address[] memory) {
```

Comment:
can break if the list becomes long but I guess it's easy to upgrade

Reply:
Yeah